CodeGeneration/Sources/generate-swiftsyntax/templates/basicformat/BasicFormatFile.swift

```diff
@@ -157,18 +157,16 @@ let basicFormatFile = SourceFile {
         switch (token.tokenKind, token.nextToken(viewMode: .sourceAccurate)?.tokenKind) {
         case (.asKeyword, .exclamationMark), // Ensures there is not space in `as!`
              (.asKeyword, .postfixQuestionMark), // Ensures there is not space in `as?`
-             (.exclamationMark, .leftParen), // Ensures there is not space in `myOptionalClosure!()`
-             (.exclamationMark, .period), // Ensures there is not space in `myOptionalBar!.foo()`
              (.initKeyword, .leftParen), // Ensures there is not space in `init()`
              (.initKeyword, .postfixQuestionMark), // Ensures there is not space in `init?`
-             (.postfixQuestionMark, .leftParen), // Ensures there is not space in `init?()`
-             (.postfixQuestionMark, .rightAngle), // Ensures there is not space in `ContiguousArray<RawSyntax?>`
-             (.postfixQuestionMark, .rightParen), // Ensures there is not space in `myOptionalClosure?()`
              (.tryKeyword, .exclamationMark), // Ensures there is not space in `try!`
              (.tryKeyword, .postfixQuestionMark): // Ensures there is not space in `try?`
           return false
         case (.spacedBinaryOperator, .comma):
           return false
+        case (.exclamationMark, .identifier), // Ensures there is not space in `try! foo`
+             (.postfixQuestionMark, .identifier): // Ensures there is not space in `try? foo`
+          return true
```

Comment:
Wouldn’t it make more sense to do 
```swift
switch (token.kind, token.parent?.kind) {
  case (.colon, .dictionaryExpr): // Covers the case at the top of this function
    return false
  case (.exclamationMark, .tryExpr), // Ensures there is a space in `try! foo`
       (.postfixQuestionMark, .tryExpr): // Ensures there is a space in `try? foo`
    return true
}
```

That should also cover cases like `try! (foo())` or `try! !functionThatThrows()`.

Reply:
Yes, exactly. Looks good!